### DISCUSSION TITLE
Rolling changelog

### DISCUSSION THREAD

**Discussion by Medard22** [Medard22/Dolphin-MMJR2-VBI#45] [Announcements]
**Draft for upcoming release**
-

Notable changes from U24.07.02-2407

- 

Missing features - these features from Dolphin Official are not implemented

- Android Input Overhaul and subsequent changes to Input and TAS handling
- Custom GPU driver support
- Skylanders portal and Infinity base support
- Gamma correction

This codebase is based on the Official Dolphin 2407-1 (Jul 2, 2024).

**General information**
- Android developers willing to help are welcome!

**Caution**

**Hints**

**Known Issues**
- The UI in the Cheats section is broken when you try to delete multiple items in a row. This bug is present to this day in Dolphin Official. Just delete a few items, exit the cheats window, and return back.
- For people using touch controls. It is possible that the core controls and touchscreen controls get out of sync. For more information check here - [https://github.com/Medard22/Dolphin-MMJR2-VBI/discussions/708](https://github.com/Medard22/Dolphin-MMJR2-VBI/discussions/708)

**Comment by ajay9634** [comment]
Thanks, now it's fixed

**Comment by ajay9634** [comment]
Hi @Medard22 

How are you?

Well I am facing Wii nunchuk touch control issues in Wii games.

The only solution is change controller to classic or other then switch back to Wii nunchuk controller.

I hope you can fix it.

**Reply by Medard22** [reply]
Hi, first off, please open a new bug report in the discussion section or post under the relevant release discussion in the future. The rolling changelog is reserved for tracking my progress between releases and isn’t intended for general support.

I’ll mark this issue as resolved and lock the discussion, but feel free to open a new bug report if you’re still unable to solve the problem.

Now I haven’t touched any code related to input, but I believe this issue happens because of a configuration override. MMJR2 launches games with the default Wiimote + Nunchuk touchscreen profile unless you switch it via Overlay Controls -> Switch Controller. It should then remember your choice unless the emulator is uninstalled or reinstalled. However, the global Wii controls configuration overrides this. So, if the Nunchuk is removed from the global configuration, the game will say it’s not connected, even though the correct touchscreen profile is displayed.

By switching to the classic controller and then back to the Nunchuk, you force the configuration to reload, which is why it starts working again.

To resolve this, make sure to delete any custom input configurations in the game profile. Then, from the main menu, click the three dots icon -> Wii Input -> Wii Remote 1 -> Emulated, and ensure that a Nunchuk extension is selected. If you don’t have any hardware buttons, simply return to the main menu and try playing your game.

This issue is one of the many reasons Dolphin’s official version proceeded with an input overhaul.
